### DISCUSSION TITLE
espanso is running but trigger/replace not working on Windows 10

### DISCUSSION THREAD

**Discussion by gnafeh** [espanso/espanso#1473] [Q&A]
OS: Windows 10 Pro

It's weird that espanso works fine on my other PC running windows 10, but I have a nuc11 that also runs windows 10 and can't trigger the replacement.
I ran espanso status and it showed that espanso is running, I tried to restart espanso and the system, but the result is still the same, espanso is running normally, just can't trigger.
The following is the espanso log.

PS C:\Users\fang\adbScripts> espanso log
15:21:37 [daemon(11232)] [INFO] reading configs from: "C:\\Users\\fang\\AppData\\Roaming\\espanso"
15:21:37 [daemon(11232)] [INFO] reading packages from: "C:\\Users\\fang\\AppData\\Roaming\\espanso\\match\\packages"
15:21:37 [daemon(11232)] [INFO] using runtime dir: "C:\\Users\\fang\\AppData\\Local\\espanso"
15:21:37 [daemon(11232)] [INFO] system info: Windows v10 (19045) - kernel: 19045
15:21:37 [daemon(11232)] [INFO] watching for changes in path: "C:\\Users\\fang\\AppData\\Roaming\\espanso"
15:21:37 [daemon(11232)] [INFO] espanso version: 2.1.8
15:21:37 [daemon(11232)] [INFO] spawning the worker process...
15:21:37 [daemon(11232)] [INFO] binded to named pipe: \\.\pipe\espansodaemonv2
15:21:37 [worker(9008)] [INFO] reading configs from: "C:\\Users\\fang\\AppData\\Roaming\\espanso"
15:21:37 [worker(9008)] [INFO] reading packages from: "C:\\Users\\fang\\AppData\\Roaming\\espanso\\match\\packages"
15:21:37 [worker(9008)] [INFO] using runtime dir: "C:\\Users\\fang\\AppData\\Local\\espanso"
15:21:37 [worker(9008)] [INFO] system info: Windows v10 (19045) - kernel: 19045
15:21:37 [worker(9008)] [INFO] binded to named pipe: \\.\pipe\espansoworkerv2
15:21:37 [worker(9008)] [INFO] using Win32AppInfoProvider
15:21:37 [worker(9008)] [INFO] monitoring the status of the daemon process
15:21:37 [worker(9008)] [INFO] using Win32Source
15:21:37 [worker(9008)] [INFO] using Win32Injector
15:21:37 [worker(9008)] [INFO] using Win32Clipboard
15:32:39 [service(5524)] [INFO] reading configs from: "C:\\Users\\fang\\AppData\\Roaming\\espanso"
15:32:39 [service(5524)] [INFO] reading packages from: "C:\\Users\\fang\\AppData\\Roaming\\espanso\\match\\packages"
15:32:39 [service(5524)] [INFO] using runtime dir: "C:\\Users\\fang\\AppData\\Local\\espanso"
15:32:39 [service(5524)] [INFO] system info: Windows v10 (19045) - kernel: 19045
15:32:39 [service(5524)] [INFO] espanso is running
PS C:\Users\fang\adbScripts>

**Comment by smeech** [comment]
Sorry about the late reply - I am trying to mop up and sort the old Issues and Discussions.

Did you ever sort this out or does it still need tagging as a bug?

**Comment by Zhrack** [comment]
Hello, I'm having the same problem.
I recently started using Espanso on my main Win10 desktop where it's working fine. 
When I installed it on my Win10 laptop, Espanso is running but nothing happens if I try any trigger.

Here is the log I have:
C:\Users\Davide>espanso log
22:42:20 [daemon(13596)] [INFO] reading configs from: "C:\\Users\\Davide\\AppData\\Roaming\\espanso"
22:42:20 [daemon(13596)] [INFO] reading packages from: "C:\\Users\\Davide\\AppData\\Roaming\\espanso\\match\\packages"
22:42:20 [daemon(13596)] [INFO] using runtime dir: "C:\\Users\\Davide\\AppData\\Local\\espanso"
22:42:20 [daemon(13596)] [INFO] system info: Windows v10 (19045) - kernel: 19045
22:42:20 [daemon(13596)] [INFO] watching for changes in path: "C:\\Users\\Davide\\AppData\\Roaming\\espanso"
22:42:20 [daemon(13596)] [INFO] espanso version: 2.2.1
22:42:20 [daemon(13596)] [INFO] spawning the worker process...
22:42:20 [daemon(13596)] [INFO] binded to named pipe: \\.\pipe\espansodaemonv2
22:42:20 [worker(11388)] [INFO] reading configs from: "C:\\Users\\Davide\\AppData\\Roaming\\espanso"
22:42:20 [worker(11388)] [INFO] reading packages from: "C:\\Users\\Davide\\AppData\\Roaming\\espanso\\match\\packages"
22:42:20 [worker(11388)] [INFO] using runtime dir: "C:\\Users\\Davide\\AppData\\Local\\espanso"
22:42:20 [worker(11388)] [INFO] system info: Windows v10 (19045) - kernel: 19045
22:42:20 [worker(11388)] [INFO] binded to named pipe: \\.\pipe\espansoworkerv2
22:42:20 [worker(11388)] [INFO] using Win32AppInfoProvider
22:42:20 [worker(11388)] [INFO] monitoring the status of the daemon process
22:42:20 [worker(11388)] [INFO] using Win32Source
22:42:20 [worker(11388)] [INFO] using Win32Injector
22:42:20 [worker(11388)] [INFO] using Win32Clipboard
22:42:21 [daemon(13596)] [INFO] configuration change detected, restarting worker process...
22:42:21 [worker(11388)] [INFO] engine eventloop has terminated, propagating exit event...
22:42:21 [worker(11388)] [ERROR] thread 'detect thread' panicked at 'unable to send to the source channel: "SendError(..)"': espanso\src\cli\worker\engine\funnel\mod.rs:104
22:42:21 [worker(11388)] [ERROR] Win32Source destruction returned non-zero code
22:42:21 [worker(11388)] [INFO] waiting for engine exit mode...
22:42:21 [worker(11388)] [INFO] exiting worker process...
22:42:22 [daemon(13596)] [INFO] spawning the worker process...
22:42:22 [worker(6004)] [INFO] reading configs from: "C:\\Users\\Davide\\AppData\\Roaming\\espanso"
22:42:22 [worker(6004)] [INFO] reading packages from: "C:\\Users\\Davide\\AppData\\Roaming\\espanso\\match\\packages"
22:42:22 [worker(6004)] [INFO] using runtime dir: "C:\\Users\\Davide\\AppData\\Local\\espanso"
22:42:22 [worker(6004)] [INFO] system info: Windows v10 (19045) - kernel: 19045
22:42:22 [worker(6004)] [INFO] binded to named pipe: \\.\pipe\espansoworkerv2
22:42:22 [worker(6004)] [INFO] using Win32AppInfoProvider
22:42:22 [worker(6004)] [INFO] monitoring the status of the daemon process
22:42:22 [worker(6004)] [INFO] using Win32Source
22:42:22 [worker(6004)] [INFO] using Win32Injector
22:42:22 [worker(6004)] [INFO] using Win32Clipboard


This is a new espanso installation, no packages are installed and no triggers get replaced.

**Comment by Geobert** [comment]
By any chance, could your issue be the same as https://github.com/espanso/espanso/issues/1582 ?

**Comment by gnafeh** [comment]
Yes, I think it’s the same issue.
In my case, I use the ‘Mouse without Borders’ to control two computers with
one keyboard and one mouse, my keyboard is physically attached to the PC A,
and espanso is installed in the PC B, so I guess that makes espanso not
working properly.

Geobert Quach ***@***.***>于2024年1月25日 周四16:11写道：

**Comment by Zhrack** [comment]
Not sure it's the same issue. I'm not using any remote control on my laptop. It's a normal installation of espanso without any plugins. The error I posted in the log is not always present, but the functionality of espanso doesn't work 100%.

**Reply by Geobert** [reply]
Do you use any other software that can generate keyboard inputs? Like KMonad, Kanata or AutoHotkey?

**Reply by Zhrack** [reply]
Nothing of that.
Altough, I'm using PowerToys to remap my broken Tab key to Alt+`. Not sure it would influence espanso keyboard detection. I'll test it by closing PowerToys this evening.

**Comment by Zhrack** [comment]
I did some tests on my Win10 laptop, it doesn't seem that PowerToys is to blame.
Posted in this txt file with Espanso log at each step. Seems like writing inside the Espanso SearchBar works, notthing else.

[espanso tests.txt](https://github.com/espanso/espanso/files/14060050/espanso.tests.txt)

For the rest, I don't have much installed, a couple games, Office suite, Obsidian for note taking and other TTRPG related software, nothing acting on keyboard input detection.
